File: backend/src/models/product.model.js
const db = require('../config/db')

/**
 * CREATE PRODUCT
 */
exports.create = (data, callback) => {
  db.run(
    `INSERT INTO products (farmer_id, name, price, stock, description)
     VALUES (?,?,?,?,?)`,
    [
      data.farmer_id,
      data.name,
      data.price,
      data.stock,
      data.description
    ],
    callback
  )
}

/**
 * GET ALL PRODUCTS (PEMBELI)
 */
exports.findAll = (callback) => {
  db.all(
    `SELECT * FROM products`,
    callback
  )
}

/**
 * GET PRODUCTS BY FARMER (PETANI)
 */
exports.findByFarmer = (farmerId, callback) => {
  db.all(
    `SELECT * FROM products WHERE farmer_id = ?`,
    [farmerId],
    callback
  )
}

File: lib/models/product_model.dart
class ProductModel {
  final int id;
  final String name;
  final int price;
  final int stock;
  final String? description;

  ProductModel({
    required this.id,
    required this.name,
    required this.price,
    required this.stock,
    this.description,
  });

  factory ProductModel.fromJson(Map<String, dynamic> json) {
    return ProductModel(
      id: json['id'],
      name: json['name'],
      price: json['price'],
      stock: json['stock'],
      description: json['description'],
    );
  }
}
